src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -178,6 +197,42 @@ private CompletableFuture<Result<LoanAndRelatedRecords>> saveCheckOutSessionReco
         succeeded(records)));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> acquireLock(LoanAndRelatedRecords records, CheckOutLockRepository checkOutLockRepository, StringBuilder checkOutLockId) {
+    log.debug("acquireLock:: Creating checkout lock {} ",records.getCheckoutLockConfiguration());
+    if(records.getCheckoutLockConfiguration() == null || !records.getCheckoutLockConfiguration().isCheckOutLockFeatureEnabled()) {
+        return completedFuture(Result.succeeded(records));
+      }
+    CompletableFuture<CheckOutLock> future = new CompletableFuture<>();
+    checkOutLockRepository.createLockWithRetry(0, future, records);
+    return future.handle((res,err) ->{
+        if(res!=null){
+          log.info("acquireLock:: Lock is acquired");
+          checkOutLockId.append(res.getId());
+          return Result.succeeded(records);
+        }
+        else{
+          log.info("acquireLock:: Unable to acquire lock");
```

Comment:
This should be `log.error` here. The message that you use should contain details from the `err` object. Don't swallow those. This message should be passed to line below as well. Don't have two messages, which is what you have now.

Reply:
Now added the err object in the logger message